src/server/logging/__tests__/applyFilterToKey.js

```diff
@@ -0,0 +1,57 @@
+var applyFilterToKey = require('../applyFilterToKey');
+var expect = require('expect.js');
+
+function fixture() {
+  return {
+    req: {
+      headers: {
+        authorization: 'Basic dskd939k2i'
+      }
+    }
+  };
+}
+
+describe('applyFilterToKey(obj, key, action)', function () {
+
+  it('should remove a key from an object recursivly', function () {
+    var data = fixture();
+    applyFilterToKey(data, 'authorization', 'remove');
+    expect(data).to.eql({
+      req: { headers: {} }
+    });
+  });
+
+  it('should remove an entire branch', function () {
+    var data = fixture();
+    applyFilterToKey(data, 'headers', 'remove');
```

Comment:
This breaks when you use `"censor"`. Maybe just remove the ability to specify removal type? Or recursively censor the entire branch?